Source/JavaScriptCore/assembler/ARMv7Assembler.h

```diff
@@ -2775,23 +2784,25 @@ class ARMv7Assembler {
 #if OS(LINUX)
         if (canBeJumpT4(reinterpret_cast<uint16_t*>(instructionStart), to)) {
             uint16_t* ptr = reinterpret_cast<uint16_t*>(instructionStart) + 2;
-            linkJumpT4(ptr, ptr, to, BranchWithLink::No);
+            linkJumpT4<jitMemcpyRepatch>(ptr, ptr, to, BranchWithLink::No);
             cacheFlush(ptr - 2, sizeof(uint16_t) * 2);
         } else {
             uint16_t* ptr = reinterpret_cast<uint16_t*>(instructionStart) + 5;
-            linkBX(ptr, ptr, to);
+            linkBX<jitMemcpyRepatch>(ptr, ptr, to);
             cacheFlush(ptr - 5, sizeof(uint16_t) * 5);
         }
 #else
         uint16_t* ptr = reinterpret_cast<uint16_t*>(instructionStart) + 2;
-        linkJumpT4(ptr, ptr, to, BranchWithLink::No);
+        linkJumpT4<jitMemcpyRepatch>(ptr, ptr, to, BranchWithLink::No);
         cacheFlush(ptr - 2, sizeof(uint16_t) * 2);
 #endif
     }
 
+    template<RepatchingInfo repatch>
     static void replaceWithNops(void* instructionStart, size_t memoryToFillWithNopsInBytes)
     {
-        fillNops<jitMemcpyRepatch>(instructionStart, memoryToFillWithNopsInBytes);
+        static_assert((*repatch).contains(RepatchingFlag::Flush));
+        fillNops<noFlush(repatch)>(instructionStart, memoryToFillWithNopsInBytes);
```

Comment:
Same detail here.  It seems the only unique piece of info we need to convey is the atomicity.  All implementations of `replaceWithNops` already require cache flushing.  So, it was superfluous to have to pass it in, and then to have to `static_assert` that it was passed in.  Is `RepatchingInfo` really needed, or can we just do with an atomicity requirement?